src/Features/Core/Portable/ExternalAccess/UnitTesting/API/NewUnitTestingIncrementalAnalyzerProvider.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.ExternalAccess.UnitTesting.SolutionCrawler;
+using Microsoft.CodeAnalysis.Host;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.UnitTesting.Api;
+
+internal sealed partial class NewUnitTestingIncrementalAnalyzerProvider : IUnitTestingIncrementalAnalyzerProvider
+{
+    private readonly string? _workspaceKind;
+    private readonly SolutionServices _services;
+    private readonly INewUnitTestingIncrementalAnalyzerProviderImplementation _incrementalAnalyzerProvider;
+
+    private IUnitTestingIncrementalAnalyzer? _lazyAnalyzer;
+
+    private NewUnitTestingIncrementalAnalyzerProvider(
+        string? workspaceKind,
+        SolutionServices services,
+        INewUnitTestingIncrementalAnalyzerProviderImplementation incrementalAnalyzerProvider)
+    {
+        _workspaceKind = workspaceKind;
+        _services = services;
+        _incrementalAnalyzerProvider = incrementalAnalyzerProvider;
+    }
+
+    // NOTE: We're currently expecting the analyzer to be singleton, so that
+    //       analyzers returned when calling this method twice would pass a reference equality check.
+    //       One instance should be created by SolutionCrawler, another one by us, when calling the
+    //       UnitTestingSolutionCrawlerServiceAccessor.Reanalyze method.
+    public IUnitTestingIncrementalAnalyzer CreateIncrementalAnalyzer()
```

Comment:
Can this be explicit interface implementation (like in the existing provider - http://index/?rightProject=Microsoft.CodeAnalysis.Features&file=ExternalAccess%5cUnitTesting%5cAPI%5cUnitTestingIncrementalAnalyzerProvider.cs&line=26)? It need not be exposed to UT...

Reply:
sure.  i can do that in followup :)